### ISSUE TITLE
Type conversion exception

### ISSUE THREAD

**Issue by LMH269084857** [influxdata/influxdb-java#791]
A field is an Integer in the database, but it is a Double value when it is retrieved

**Comment by UglyHobbitFeet** [comment]
YES!, this is happening to me as well. I have the latest version installed (2.22)

When I use chronograf and look at the data it is indeed of type integer but has a value of decimal

FWIW I'm creating the record the line protocol via:
````
this.influxDb.write(myInfluxDbName, null, ConsistencyLevel.ANY, myLineProtocolData);
````

I am also dynamically creating this on the fly via the line of code shown above. In other words, I did not create the measurement on startup/creation before running the above line of code.

As a temp workaround I am casting the result from double to int, but I shouldnt have to be doing that

**Comment by UglyHobbitFeet** [comment]
Not sure if this is the reason, but it looks like it might be why (Line 157)
https://github.com/influxdata/influxdb-java/blob/master/src/main/java/org/influxdb/dto/Point.java

**Comment by LMH269084857** [comment]
> Not sure if this is the reason, but it looks like it might be why (Line 157) https://github.com/influxdata/influxdb-java/blob/master/src/main/java/org/influxdb/dto/Point.java#L157

Yes, I've tried other types of integer data. It does convert all integer data to double

**Comment by UglyHobbitFeet** [comment]
This has been a major blocker for a while now!  Is the author of this code able to fix this?

**Comment by majst01** [comment]
If someone is willing to look deeper into the the root cause of this i will review any PR.

**Comment by UglyHobbitFeet** [comment]
@majst01 I already did. It's mentioned on this thread a few comments up :)

**Comment by majst01** [comment]
Sure, this is the why, but whats the solution ?

**Comment by UglyHobbitFeet** [comment]
@majst01 To change  "value = ((Integer) value).doubleValue()"; to  "value = ((Integer) value)";

**Comment by majst01** [comment]
> @majst01 To change "value = ((Integer) value).doubleValue()"; to "value = ((Integer) value)";

Fine, write a PR and make all unit tests pass again

**Comment by UglyHobbitFeet** [comment]
@majst01 Unfortunately I don't know how to do that. I'm not a big githubber sorry

**Comment by majst01** [comment]
> @majst01 Unfortunately I don't know how to do that. I'm not a big githubber sorry

https://jarv.is/notes/how-to-pull-request-fork-github/

**Comment by UglyHobbitFeet** [comment]
@majst01 I appreciate the links.  I tried but unfortunately I shouldn't be the one doing PR as I'm not familiar enough with the overall codebase and the impact of changing existing test cases.  I may end up messing a useful test case to 'fit' my fix.  I know what the problem is and have pointed to the line of code that is causing it... that's the best I can do.
